backend/de.metas.acct.base/src/main/java-legacy/org/compiere/acct/DocLine_Allocation.java

```diff
@@ -405,6 +406,18 @@ public BPartnerId getInvoiceBPartnerId()
 		return getBPartnerId();
 	}
 
+	@Nullable
+	public BPartnerLocationId getInvoiceBPartnerLocationId()
+	{
+		final I_C_Invoice invoice = getC_Invoice();
+		if (invoice != null)
+		{
+			return BPartnerLocationId.ofRepoId(invoice.getC_BPartner_ID(), invoice.getC_BPartner_Location_ID());
+		}
+
+		return getBPartnerLocationId();
```

Comment:
why not return null if there is no invoice?

Reply:
I was just preserving the logic from C_BPartner_ID

Reply:
imho makes no sense to fallback